.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,77 @@
+# Release vFIXME
+
+## Checklist
+
+- [ ] [Create a new github milestone](https://github.com/DataDog/dd-trace-go/milestones/new) for the next version and
+  move any open PRs and issues that are in the current release milestone to the new one.
+
+- [ ] Draft [the release note](https://github.com/DataDog/dd-trace-go/releases/new) and ask the teams for its review on
+  the `#guild-dd-go` channel.
+
+- Deploy a release candidate:
+    - Create a release candidate version on this release branch:
+        - [ ] Bump and commit the release candidate version string in `/internal/version/version.go` (eg. `v1.2.3-rc.4`)
+        - [ ] Tag the bump commit with the release candidate version (eg. `git tag v1.2.3-rc.4`)
+        - [ ] Push the tag and commit (eg. `git push --tags origin release-v1.2.3`)
```

Comment:
I think we could avoid that now that we can directly and simply `go get -u .../dd-trace-go@release-v1.2.3`. This would be a huge simplification of this process.

Reply:
The only downside of this is that the reported version wouldn't have the -rc part in the traces.

Reply:
Are you saying that we wouldn't need to push the tags in that case? I don't understand the suggestion here.

Reply:
I meant that you wouldn't change the version.go file nor git tag the latest commit, you would only `go get dd-trace-go@release-v1.2.3` for example in order to update a `go.mod` to use the latest release branch commit instead.

But because we currently track our deployments with the dd-trace-go version in version.go, I think we should continue doing individual RC versions and tags so that we can easily identify them on our dashboards.

cc @felixge